pdf/plugin-dowker/main.py

```diff
@@ -0,0 +1,165 @@
+            matrix.append({})
+          #  lines.append(line)
+            i=1
+            a="pdfinfo_<>"
+            matrix[-1][0]=file_names[-1]
+            curr+=1
+          #  errorMatrix.resize(len(ft_names),curr, False)
+            
+            for name in ft_names:
+                matrix[-1][i]=0
+                errorMatrix[curr,i]=1
+                matched =False
+                for key in obj.keys():
+                    if name.split("\\<\\<")[0]==key.split("\\<\\<")[0]:
+                        matrix[-1][i]=obj[key]
+                        errorMatrix[curr,i]=1-int(obj[key])
+                        matched=True
+                        break
+                if not(matched):
+                    errorMatrix[curr, i]=1
+           #     matrix[-1][i]="|"+list(obj.keys())[0] +"|"
+               # matrix[-1][i]=obj[list(obj.keys())[0]]
+               # if name in obj:
+                #    matrix[-1][i]=obj[name]
+               # else:
+                #    matrix[-1][i]=name
+              #  matrix[-1][i]=line[0:300]
+                i+=1
+#            for k, v in obj.items():
+ #               print(k)
+                
+        elif mode == 'refs':
+            # Ensure UI maintains its old decision (this was a regression; the
+            # new code does not have this flaw)
+            #print(line)
+            lines.append(line)
+            if not use_refs:
+                continue
+
+            obj = json.loads(line)
+            if obj['status'] == 'valid':
+                set1.add(obj['testfile'])
+            else:
+                set2.add(obj['testfile'])
+        else:
+            raise NotImplementedError(mode)
+    file1.close()
+	#Create Dowker from matrix
+    errorMatrixToDowker(errorMatrix)
+	#Return plotly args
+	
+
+    # Sort files based on likelihood of being 'odd'
+
+    html_vars['lines']=str(errorMatrix)
+    write_html(os.path.join(_path, 'index.pug'), html_vars, args.html_out)
+
+
+def write_html(template_file, vars, html_out):
+    """Given a template in `template_file`, format it in the Pug language
+    using `vars` as context variables, printing the resulting HTML to stdout.
+    """
+    with open(template_file) as f:
+        template = f.read()
+
+    parser = pypugjs.parser.Parser(template)
+    block = parser.parse()
+    compiler = pypugjs.ext.html.Compiler(block)
+    if True:#packaging.version.parse(pypugjs.__version__) < packaging.version.parse('5.9.9'):
+        # Fix 5.9.4 and before
+        def interpolate_replacement(self, text, escape=True):
+            esc = lambda x: x
+            if escape:
+                esc = lambda x: x.replace('&', '&amp;').replace('<', '&lt;').replace('>', '&gt;')
+            return self._interpolate(text, lambda x: esc(str(self._do_eval(x))))
+        pypugjs.ext.html.Compiler.interpolate = interpolate_replacement
+    compiler.global_context = vars
+    with open("dowker_faw.html", 'r') as f1:
```

Comment:
It looks like you've integrated a single, pre-compiled Dowker plot. From what I can tell, this won't display the output for the files being inspected by the FAW, though. I suspect you need to pass `vars['lines']` somewhere based on the integration of `errorMatrixToDowker`